stellar/deploy-contract.js

```diff
@@ -96,36 +101,83 @@ async function processCommand(options, config, chain) {
     await broadcast(operation, wallet, chain, 'Initialized contract', options);
 }
 
+async function processUpgrade(options, _, chain) {
+    const { contractName, contractId, privateKey, wasmPath, newWasmHash } = options;
+    const { rpc, networkType } = chain;
+    const networkPassphrase = getNetworkPassphrase(networkType);
+
+    if (!chain.contracts) {
+        chain.contracts = {};
+    }
+
+    const args = `--source ${privateKey} --rpc-url ${rpc} --network-passphrase "${networkPassphrase}"`;
+
+    let cmd;
+
+    if (options.install) {
+        cmd = `${stellarCmd} contract install --wasm ${wasmPath} ${args}`;
+    } else {
+        cmd = `${stellarCmd} contract invoke --id ${contractId} ${args} -- upgrade --new_wasm_hash ${newWasmHash}`;
+    }
```

Comment:
do both when upgrading to make it easier to use. the install step just uploads the bytecode so it's a simple step (if bytecode was already uploaded before, it's a no-op on-chain I believe)